tests/specs/functional/lemmas-spec.k

```diff
@@ -0,0 +1,50 @@
+requires "../lemmas.k"
+
+module VERIFICATION
+    imports LEMMAS
+
+    syntax StepSort ::= Map | WordStack
+ // -----------------------------------
+
+    syntax KItem ::= runLemma ( StepSort )
+                   | doneLemma( StepSort )
+ // --------------------------------------
+    rule runLemma( T ) => doneLemma( T )
+
+endmodule
+
+module LEMMAS-SPEC
+    imports VERIFICATION
+
+ // Buffer write simplifications
+ // ----------------------------
+
+    rule <k> runLemma  ( M [ L := .WordStack ] [ N := WS:WordStack ]                             ) => doneLemma ( M [ N := WS ]                        ) ... </k>
+    rule <k> runLemma  ( M [ L := WS:WordStack ] [ N := .WordStack ]                             ) => doneLemma ( M [ L := WS ]                        ) ... </k>
+    rule <k> runLemma  ( M [ 32 := (0 : 1 : 2 : .WordStack) ] [ 32 := (3 : 4 : 5 : .WordStack) ] ) => doneLemma ( M [ 32 := (3 : 4 : 5 : .WordStack) ] ) ... </k>
+
+ // #range selection operation
+ // --------------------------
+
+    // Non-positive range
+    rule <k> runLemma ( #range(M, N,  0) ) => doneLemma ( .WordStack ) ... </k>
+    rule <k> runLemma ( #range(M, N, -3) ) => doneLemma ( .WordStack ) ... </k>
+
+    // Range outside of previous buffer write
+    rule <k> runLemma ( #range(M [ 32 := BUF:WordStack ], 20, 4) ) => doneLemma ( #range(M, 20, 4) ) ... </k>
+    rule <k> runLemma ( #range(M [ 32 := BUF:WordStack ], 48, 4) ) => doneLemma ( #range(M, 48, 4) ) ... </k> requires #sizeWordStack(BUF) <=Int 12
```

Comment:
Please add the following corner cases (ranges of size 1, right before and right after the previous buffer):
```
rule <k> runLemma ( #range(M [ 32 := BUF:WordStack ], 31, 1) ) => doneLemma ( #range(M, 31, 1) ) ... </k>
rule <k> runLemma ( #range(M [ 32 := #buf(1, DATA) ], 33, 1) ) => doneLemma ( #range(M, 33, 1) ) ... </k>
```

Reply:
Added!